packages/backend/src/plugins/search.ts

```diff
@@ -68,7 +68,7 @@ export default async function createPlugin({
   // particular collator gathers entities from the software catalog.
   indexBuilder.addCollator({
     defaultRefreshIntervalSeconds: 600,
-    collator: new DefaultCatalogCollator({ discovery }),
+    collator: DefaultCatalogCollator.fromConfig(config, { discovery }),
```

Comment:
I think you can just refer to type as `CatalogEntitiesRequest['filter']` if you import `CatalogEntitiesRequest` from `@backstage/catalog-client`. And yeah use that type directly, don't have to make some special filter syntax for your own class that then gets translated to the catalog client syntax

Reply:
One more thing, sorry to be such a pest.

When making changes to the app, it sometimes makes sense to make the same change in the create-app template so that future adopters also get the benefit of the same changes. In your case the change _happened_ to be backwards compatible (an optional parameter, and didn't make the constructor private), but check this out

https://github.com/backstage/backstage/blob/1fa98590a9629e8e9dc399b21f44a20139a38497/packages/create-app/templates/default-app/packages/backend/src/plugins/search.ts?plain=1#L22

And then add a changeset for `@backstage/create-app` telling people what they need to change in their own app to conform to the new preferred way of initializing the class. See the create-app changelog for inspiration on how we sometimes write these.